lib/checkbool.cpp

```diff
@@ -454,3 +455,27 @@ void CheckBool::assignBoolToFloatError(const Token *tok)
     reportError(tok, Severity::style, "assignBoolToFloat",
                 "Boolean value assigned to floating point variable.", CWE704, false);
 }
+
+void CheckBool::returnValueOfFunctionReturningBool(void)
+{
+    if (!mSettings->isEnabled(Settings::STYLE))
+        return;
+
+    const SymbolDatabase * const symbolDatabase = mTokenizer->getSymbolDatabase();
+
+    for (const Scope * scope : symbolDatabase->functionScopes) {
+        if (!(scope->function && Token::Match(scope->function->retDef, "bool|_Bool")))
+            continue;
+
+        for (const Token* tok = scope->bodyStart->next(); tok && (tok != scope->bodyEnd); tok = tok->next()) {
+            if (Token::simpleMatch(tok, "return") &&
```

Comment:
I think you should skip lambdas in the function.

Using `findLambdaEndToken` might be a good idea. However it seems to me this function should handle `-> ..` better. You do not need to add such handling in this PR though.

Reply:
Good catch, lambdas in a function returning bool was giving false positives. It still does if the return type of the lambda is explicitly specified with `-> ..` (like you pointed out). I added TODO-testcases for that. It doesn't look too difficult to make `findLambdaEndToken` recognize `-> ..`, I can take a look at that, but perhaps for a different PR? When that's done, perhaps this check could be extended to look for `return` in lambdas as well?